lib/auth/authProvider.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+const MongoCredentials = require('./mongoCredentials').MongoCredentials;
+const MongoError = require('../error').MongoError;
+
+/**
+ * Creates a new Authentication mechanism
```

Comment:
I'm being a bit of a nitpick on documentation, but I think it should be very clear what the design is. Can you provide more color on what an Authentication Mechanism is, and what this base class is meant do.